File: C++/Ders/Ders-4/Student.cpp
//
// Created by jabru on 8/29/2022.
//

#include "Student.h"

#include <utility>

Student::Student(const string &name, const string &surname, int age, string department) : Person(name, surname,
                                                                                                        age),
                                                                                                 department(std::move(
                                                                                                         department)) {}

const string &Student::getDepartment() const {
    return department;
}

void Student::setDepartment(const string &department) {
    Student::department = department;
}

ostream &operator<<(ostream &os, const Student &student) {
    os << static_cast<const Person &>(student) << "\ndepartment: " << student.department;
    return os;
}

File: JAVA/Ders/Ders-2/src/OOP1/Student.java
package OOP1;

public class Student {
    private String name; //null
    private String surname; //null
    private int ID; // 0

    Student(){

    }

    Student(String name, String surname, int ID){
        this.ID = ID;
        this.name = name;
        this.surname = surname;
    }

    @Override
    public String toString() {
        return "Student{" +
                "name='" + name + '\'' +
                ", surname='" + surname + '\'' +
                ", ID=" + ID +
                '}';
    }
}
